src/app.tsx

```diff
@@ -1,15 +1,26 @@
 declare let window: any;
-import React, { FunctionComponent } from "react";
+import React, { FunctionComponent, useEffect, useState } from "react";
 import "./app.css";
 import { MainPage } from "./page/main-page";
 import { MetaMaskError } from "./page/metamask-error";
+import { MetaMaskPending } from "./page/metamask-pending";
 
 const App: FunctionComponent = () => {
+  const [metamaskConnected, setMetamaskConnected] = useState(false);
+  useEffect(() => {
+    checkMetaMaskAccount();
+  });
+
+  const checkMetaMaskAccount = async () => {
+    const accounts = await window.ethereum.request({ method: "eth_requestAccounts" });
+    accounts.length > 0 ? setMetamaskConnected(true) : setMetamaskConnected(false);
+  };
+
   if (!window.ethereum) {
     return <MetaMaskError />;
   }
 
-  return <MainPage />;
+  return metamaskConnected ? <MainPage /> : <MetaMaskPending />;
```

Comment:
great. If I click on cancel on metamask, the pending page stays.

Reply:
done. I added a check for code 4001 and bring users to the main page when they cancel the metamask.

Reply:
![image](https://user-images.githubusercontent.com/503185/125250515-ebd90f00-e328-11eb-8405-e6fca46f2e7a.png)

Should only display the connect button

Reply:
Done.